packages/playwright-core/src/server/registry/index.ts

```diff
@@ -445,14 +479,32 @@ export class Registry {
       executablePath: () => chromiumExecutable,
       executablePathOrDie: (sdkLanguage: string) => executablePathOrDie('chromium', chromiumExecutable, chromium.installByDefault, sdkLanguage),
       installType: chromium.installByDefault ? 'download-by-default' : 'download-on-demand',
-      _validateHostRequirements: (sdkLanguage: string) => this._validateHostRequirements(sdkLanguage, 'chromium', chromium.dir, ['chrome-linux'], [], ['chrome-win']),
+      _validateHostRequirements: (sdkLanguage: string) => this._validateHostRequirements(sdkLanguage, chromium.dir, ['chrome-linux'], [], ['chrome-win']),
       downloadURLs: this._downloadURLs(chromium),
       browserVersion: chromium.browserVersion,
       _install: () => this._downloadExecutable(chromium, chromiumExecutable),
       _dependencyGroup: 'chromium',
       _isHermeticInstallation: true,
     });
 
+    const chromiumHeadlessShellDescriptor = { ...chromium, name: 'chromium-headless-shell', dir: chromium.dir.replace(/(.*)(-\d+)$/, '$1-headless-shell$2') };
```

Comment:
To make it explicit and expose issues like wrong `installByDefault` value.

```suggestion
    const chromiumHeadlessShellDescriptor: BrowsersJSONDescriptor = {
      name: 'chromium-headless-shell',
      revision: chromium.revision,
      browserVersion: chromium.browserVersion,
      dir: chromium.dir.replace(/(.*)(-\d+)$/, '$1-headless-shell$2'),
      installByDefault: false
    };
```